pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -764,13 +748,16 @@ void PdBlock_iterBegin(JSOBJ _obj, JSONTypeContext *tc) {
     if (!blocks) {
         GET_TC(tc)->iterNext = NpyArr_iterNextNone;
         goto BLKRET;
+    } else if (!PyTuple_Check(blocks)) {
+      PyErr_SetString(PyExc_TypeError, "blocks must be a tuple!");
+      goto BLKRET;
     }
 
     // force transpose so each NpyArrContext strides down its column
     GET_TC(tc)->transpose = 1;
 
     for (i = 0; i < PyObject_Length(blocks); i++) {
-        block = get_item(blocks, i);
+        block = PyTuple_GET_ITEM(blocks, i);
```

Comment:
nice cleanup

Reply:
The type hints in internals were very helpful to notice this, so thank you for the work you've been doing there